src/domain/community/community/community.resolver.mutations.ts

```diff
@@ -189,6 +186,51 @@ export class CommunityResolverMutations {
     return savedApplication;
   }
 
+  @UseGuards(GraphqlGuard)
+  @Mutation(() => ICommunity, {
+    description: 'Join the specified Community as a member.',
+  })
+  @Profiling.api
+  async joinCommunity(
+    @CurrentUser() agentInfo: AgentInfo,
+    @Args('joinCommunityData') joiningData: CommunityJoinInput
```

Comment:
if only a user is allowed to join by themselfs, why is an userID required in the input?

Reply:
Good point. I have adjusted both mutations to remove the userID from the input type, and the mutation now picks that up from the agentInfo

Reply:
What is the advantage of using an input object over an inline argument just for the `communityID`?